internal/handlers/middleware.go

```diff
@@ -2,12 +2,16 @@ package handlers
 
 import (
 	"net/http"
+	"regexp"
 	"strings"
 	"time"
 
+	"github.com/google/uuid"
 	"github.com/rs/cors"
 )
 
+var initrdPattern = regexp.MustCompile(`^/images/(?P<uuid>.*)/pxe-initrd`)
```

Comment:
That would make matching a bit more complex - and, more importnantly, hardcode UUID when InfraEnv ID format is encoded in assisted-service.

Perhaps we should create "isIDValid" for infraenvs and reuse it here? If that gets too complicated we can drop this check

Reply:
I don't have a problem breaking if assisted service all of a sudden decides not to use uuids anymore. Especially because we're already validating the uuid format.

Reply:
Ah, okay. imo the regexp would work slower that `uuid.Parse`, but otoh it would make code look better, so lets use that